app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -86,7 +86,14 @@ EOF
     echo "Building yum metadata cache, this may take a few minutes"
     $PKG_MANAGER makecache
   else
-    zypper --quiet --non-interactive addrepo /tmp/foreman_registration.repo
+    REPO_OPTIONS=""
+    <% if repo_gpg_key_url.present? -%>
+    <%= generate_web_request(utility: @download_utility, url: shell_escape(repo_gpg_key_url)) %> >/tmp/repo_gpg.key
+    rpm --import /tmp/repo_gpg.key
+    rm -f /tmp/repo_gpg.key
+    REPO_OPTIONS='--gpgcheck-allow-unsigned-repo'
```

Comment:
Is this wanted? I would expect opposite behavior, if the `repo_gpg_key_url` is present, the GPG check should happen, and the repo should be signed with it.

Reply:
Yes, it is intended. 

The repository metadata is unsigned (because pulp does not support to sign the metadata).

But the packages within the repo are signed.

In Suse, normally both is signed: metadata and packages itself.

Reply:
Ah, okay, thanks for the explanation.